custom_components/tuya_local/devices/gorilla_fan_v2.yaml

```diff
@@ -0,0 +1,58 @@
+name: Gorilla Fan V2
+primary_entity:
+  entity: fan
+  dps:
+    - id: 1
+      type: boolean
+      name: switch
+      icon: mdi:ceiling-fan
+    - id: 2
+      type: string
+      name: preset_mode
+      mapping:
+        - dps_value: normal
+          value: Normal
+        - dps_value: boost
+          value: Boost
+    - id: 3
+      type: integer
+      name: speed
+      range:
+        min: 1
+        max: 5
+
+secondary_entities:
+  - entity: light
+    name: Night Light
+    icon: mdi:ceiling-fan-light
+    dps:
+      - id: 15
+        type: boolean
+        name: switch
+  - entity: switch
+    name: Boost
```

Comment:
What is the difference between this and the mode (dp 2)?
It seems confusing to have both named the same way.

Reply:
dp 2 changes when the switch for boost changes but the vice-versa doesn't happen, I'll remove dp2

Reply:
Then I think you can move this into the fan as preset_mode (mapping: false -> normal, true -> strong) and add `translation_key: fan_with_presets` to the fan, to get those translated.  Rename the dp 2 to something not standard in the fan entity (probably the tuya naming is ok) to leave it as an extra attribute for reference (probably without any mapping is best, since it is only for reference).

Reply:
I attempted the suggested changes, and while they were beneficial, I encountered an issue. When changing the speed, the boost mode should revert to normal, but this isn't happening. Additionally, I believe it's more convenient to maintain it as a separate entity.
Although boost functions as a 6th speed, the speed value does not change when set to boost.

Reply:
The issue you encountered is not really resolved by moving Boost from a preset_mode of the fan to a separate switch.  The way to solve it would be to combine it with speed:

```
    - id: 3
      type: integer
      name: speed_setting
      hidden: true
    - id: 115
      name: speed
      type: boolean
      mapping:
        - dps_val: true
          value: 100
        - dps_val: false
          constraint: speed_setting
          conditions:
            - dps_val: 1
              value: 17
            - dps_val: 2
              value: 33
            - dps_val: 3
              value: 50
            - dps_val: 4
              value: 67
            - dps_val: 5
              value: 83
```